src/main/scala/za/co/absa/spark_partition_sizing/RowSizer.scala

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2021 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.spark_partition_sizing
+
+import org.apache.spark.sql.Row
+import org.apache.spark.sql.catalyst.expressions.GenericRowWithSchema
+import org.apache.spark.util.SizeEstimator
+import za.co.absa.spark_partition_sizing.types.ByteSize
+
+
+object RowSizer {
+
+  def rowSize(row: Row)(implicit fieldNames: Seq[String]): ByteSize = {
```

Comment:
Nice to have an abstraction, where the implementation can change.
But I would wary to have such a common type as `Seq[String]` as `implicit` parameter, it's borderline imho.